apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php

```diff
@@ -109,6 +109,44 @@ function add_coming_soon_page_to_new_site() {
 }
 add_action( 'signup_finished', __NAMESPACE__ . '\add_coming_soon_page_to_new_site', 10, 1 );
 
+/**
+ * Launch the site when the privacy mode changes from public-not-indexed
+ * This can happen due to clicking the launch button from the banner
+ * Or due to manually updating the setting in wp-admin or calypso settings page.
+ *
+ * @param string $old_value the old value of blog_public.
+ * @param string $value     the new value of blog_public.
+ */
+function disable_coming_soon_on_privacy_change( $old_value, $value ) {
+	if ( 0 !== (int) $old_value || 0 === (int) $value ) {
+		// Do nothing if not moving from public-not-indexed.
+		return;
+	}
+	update_option( 'wpcom_public_coming_soon', 0 );
+}
+add_action( 'update_option_blog_public', __NAMESPACE__ . '\disable_coming_soon_on_privacy_change', 10, 2 );
```

Comment:
do we need this?

Reply:
> do we need this?

I think we do. It's trying to solve what I've done in D50760-code (see `launch_site_on_privacy_change()`, but in the plugin, which I think is a good place for it.

D50760-code still ensures that we're setting the right launch status for new, public + Coming Soon sites (see `wpmu_new_blog_set_launch_status()`) - is that possible to do in the plugin as well so we can bypass those freaky conditions in the backend?